python/src/aiconfig/scripts/aiconfig_cli.py

```diff
@@ -122,6 +152,30 @@ def is_port_in_use(port: int) -> bool:
         return s.connect_ex(("localhost", port)) == 0
 
 
+def _start_server(
+    start_config: StartServerConfig, aiconfigrc_path: str
+) -> Result[list[str], str]:
+    port = start_config.server_port
+
+    if is_port_in_use(port):
+        err = f"Port {port} is in use. Cannot start server."
+        LOGGER.error(err)
+        return Err(err)
+
+    LOGGER.warning(f"Using {port} to start aiconfig server.")
+
+    results: list[Result[str, str]] = []
+    backend_res = run_backend_server(start_config, aiconfigrc_path)
+    match backend_res:
+        case Ok(_):
+            pass
+        case Err(e):
+            return Err(e)
```

Comment:
nit: I don't think you need this block. You'll return the natural Ok or Err on line 176